File: src/algorithm1/day4/ReverseWordsInAString3.java
package algorithm1.day4;

public class ReverseWordsInAString3 {
    public static void main(String[] args) {
        String s = "Let's take LeetCode contest";

        System.out.println("reverseWords(s) = " + reverseWords(s));
    }

    public static String reverseWords(String s) {
        String[] strings = s.split(" ");
        StringBuilder sb = new StringBuilder();

        for (String string : strings) {
            sb.append(reverse(string.toCharArray()) + " ");
        }

        String result = sb.toString();
        result = result.substring(0, result.length() - 1);
        return result;
    }

    public static String reverse(char[] chars) {
        int left = 0;
        int right = chars.length - 1;

        while (left < right) {
            char temp = chars[left];
            chars[left] = chars[right];
            chars[right] = temp;
            left++;
            right--;
        }

        return String.valueOf(chars);
    }
}

File: src/algorithm1/kotlinsource/day4/ReverseWordsInAString3.kt
package algorithm1.kotlinsource.day4

class ReverseWordsInAString3 {
    fun reverseWords(s: String): String {
        var strings = s.split(" ")
        var sb = StringBuilder()

        for (index in strings.indices) {
            val reverseChars = reverse(strings.get(index).toCharArray())
            sb.append(reverseChars).append(" ")
        }

         val result = sb.toString()
        return result.substring(0, result.lastIndex)
    }

    private fun reverse(chars: CharArray): CharArray {
        var left = 0
        var right = chars.lastIndex

        while (left < right) {
            var temp = chars[left]
            chars[left] = chars[right]
            chars[right] = temp
            left += 1
            right -= 1
        }

        return chars
    }
}